code/modules/mob/living/life.dm

```diff
@@ -1,57 +1,27 @@
 /// This divisor controls how fast body temperature changes to match the environment
 #define BODYTEMP_DIVISOR 8
 
-/mob/living/proc/Life(seconds, times_fired)
-	set waitfor = FALSE
-
-	if((movement_type & FLYING) && !(movement_type & FLOATING))	//TODO: Better floating
-		float(on = TRUE)
-
-	if (client)
-		var/turf/T = get_turf(src)
-		if(!T)
-			move_to_error_room()
-			var/msg = "[ADMIN_LOOKUPFLW(src)] was found to have no .loc with an attached client, if the cause is unknown it would be wise to ask how this was accomplished."
-			message_admins(msg)
-			send2tgs_adminless_only("Mob", msg, R_ADMIN)
-			log_game("[key_name(src)] was found to have no .loc with an attached client.")
-
-		// This is a temporary error tracker to make sure we've caught everything
-		else if (registered_z != T.z)
-#ifdef TESTING
-			message_admins("[ADMIN_LOOKUPFLW(src)] has somehow ended up in Z-level [T.z] despite being registered in Z-level [registered_z]. If you could ask them how that happened and notify coderbus, it would be appreciated.")
-#endif
-			log_game("Z-TRACKING: [src] has somehow ended up in Z-level [T.z] despite being registered in Z-level [registered_z].")
-			update_z(T.z)
-	else if (registered_z)
-		log_game("Z-TRACKING: [src] of type [src.type] has a Z-registration despite not having a client.")
-		update_z(null)
-
-	if (notransform)
-		return
-	if(!loc)
-		return
-
-	if(!IS_IN_STASIS(src))
-
-		if(stat != DEAD)
-			//Mutations and radiation
-			handle_mutations_and_radiation()
-
-		if(stat != DEAD)
-			//Breathing, if applicable
-			handle_breathing(times_fired)
+/**
+  * Handles all mob updates
+  */
+/mob/living/proc/life_process()
+	SHOULD_NOT_SLEEP(TRUE)
+	SHOULD_CALL_PARENT(TRUE)
```

Comment:
We are going to need a way to have our cake and eat it too.

I get wanting to lint sleeps, but i can't support removing `set waitfor = FALSE`

the linter is not perfect, when it comes to procs called via call() or callbacks this system can not know that something sleeps. nor can it detect cases where type confusion means a proc call happens on a type different then what the linter expects.

If life_process() *does* sleep, despite this check, it ***must not*** hold up ssmob.